### PR TITLE
Cs jump sounds

### PR DESCRIPTION
**Pull request by drzel**

### CODE REVIEW

File: csqc/main.qc [hunk 1]
```diff
@@ -174,17 +176,19 @@ noref void(float width, float height, float menushown) CSQC_UpdateView = {
 
     local float paused = getstatf(STAT_PAUSED);
 
-    if (pmove_onground && health > 0 && !paused) {
-        // The pmove_vel_z < 180 check makes sure you're not sliding
-        if (input_buttons & BUTTON2 && pmove_vel_z < 180) {
-            localsound("player/plyrjmp8.wav", CHAN_BODY, cvar(FOCMD_JUMPVOLUME));
-        }
+    if (cvar("csqcjumpsounds")) {
```
Comment (newbytf) [inline csqc/main.qc]: Can you mix cvar() and autocvar?  Even if it works, you're giving up all the benefits of autocvar here.

File: csqc/events.qc [hunk 2]
```diff
@@ -444,6 +444,9 @@ void ParseSBAR()
 }
 
 float(float entnum, float channel, string soundname, float vol, float attenuation, vector pos, float pitchmod) CSQC_Event_Sound = {
+    if !(cvar("csqcjumpsounds"))
+        return 0;
```
Comment (newbytf) [inline csqc/events.qc:448]: Put this into the switch so that it only applies to the jump sound cases?

File: csqc/main.qc [hunk 3]
```diff
@@ -131,6 +133,30 @@ noref void(float width, float height, float menushown) CSQC_UpdateView = {
...
+
+            if (!last_pmove_onground) {
+                if (last_vel_z < -650) {
+                    localsound("player/land2.wav", CHAN_AUTO, 1);
+                } else if (last_vel_z < -300) {
+                    localsound("player/land.wav", CHAN_AUTO, 1);
+                }
+            }
+        }
+    }
```
Comment (drzel) [inline csqc/main.qc:156]: Moved back to `CSQC_UpdateView` since `pmove_vel_z` evaluating to `0` atm.

File: csqc/main.qc (hunk 1 above)
Comment (drzel) [reply csqc/main.qc]: You can but I've updated all to use autocvar.

File: csqc/events.qc (hunk 2 above)
Comment (drzel) [reply csqc/events.qc]: Done.

File: csqc/main.qc (hunk 3 above)
Comment (newbytf) [reply csqc/main.qc]: This breaks the whole point of the optimization -- moving it out of the render path.  If pmove_vel_z isn't set properly in input frame (makes sense, given ordering with pmove) then just  cache only the value it in update_view but leave the rest here and I'll get it out of update view again once I start touching pmove.

File: csqc/main.qc [hunk 4]
```diff
@@ -132,7 +132,7 @@ noref void(float width, float height, float menushown) CSQC_UpdateView = {
     Hud_Draw(width, height);
     sui_end();
 
-    if (cvar("fo_csjumpsounds")) {
+    if (autocvar(fo_csjumpsounds, 1, "client-side predicted jump sounds")) {
```
Comment (newbytf) [inline csqc/main.qc:135]: Commits merged to master last night added CVARF accessors for this.

File: csqc/events.qc [hunk 5]
```diff
@@ -444,17 +444,16 @@ void ParseSBAR()
...
         return 0;
 
     switch(soundname) {
         case "player/plyrjmp8.wav":
         case "player/land.wav":
         case "player/land2.wav":
-            return 1;
+        case "player/h2ojump.wav":
+            if (autocvar(fo_csjumpsounds, 1))
+                return 1;
```
Comment (newbytf) [inline csqc/events.qc:458]: Need a break here for the else case.

File: csqc/main.qc (hunk 4 above)
Comment (drzel) [reply csqc/main.qc]: Done.

File: csqc/events.qc (hunk 5 above)
Comment (drzel) [reply csqc/events.qc:458]: Heh thanks for this. Just pretend I don't know anything about c and you won't be far off the mark.

**Pull request merged**